File: jlc_ruby/example/belly.clj
(ns example.belly
    (:require [fundingcircle.jukebox :refer [pending! step]]))

(defn- helper-fn-a
       "A test helper fn."
       {:scene/resources [:kafka/topic-a]}
       [])

(defn- helper-fn-b
       "A test helper fn."
       {:scene/resources [:kafka/topic-b]}
       [])

(defn- helper-fn-c
       "A test helper fn."
       {:scene/resources [:kafka/topic-c]}
       [])

(defn- helper-fn-d
       "A test helper fn."
       {:scene/resources [:kafka/topic-d]}
       []
       (helper-fn-c))

(defn- helper-fn-e
       "A test helper fn."
       []
       '(helper-fn-e))

(defn ^:scene/before before-scenario
      "Called before a scenario."
      {:scene/resources [:kafka/topic-d :kafka/topic-f]
       :scene/tags "@bat"}
      [board _scenario]
      (println "BEFORE SCENARIO 1")
      board)

(defn ^:scene/before-step before-step
      "Called before each step in a scenario."
      {:scene/resources [:kafka/topic-d :kafka/topic-f]
       :scene/tags "@bat"}
      [board _scenario]
      board)

(defn ^:scene/after-step after-step
      "Called after each step in a scenario."
      {:scene/resources [:kafka/topic-d :kafka/topic-f]
       :scene/tags "@bat"}
      [board _scenario]
      board)

(defn i-have-cukes-in-my-belly
      {:scene/step "I have {int} cukes in my belly"
       :scene/resources [:kafka/topic-e]}
      [board number-of-cukes]
      (helper-fn-a)
      (helper-fn-b)
      (helper-fn-d)
      (helper-fn-e)
      board)

(step "I have this table"
      [board data-table]
      (assoc board :table data-table))

(defn ^:scene/after to-run-after-scenario
      {:scene/tags "@qux"}
      [board _scenario]
      board)

(step :before {:tags "@foo or @bar and @clj"}
      [board scenario]
      board)

(step "a clojure step that fails"
      [board]
      (pending!) ;; Write code here that turns the phrase above into concrete actions
      board) ;; Return the updated board

(step :after_step
      [board scenario]
      board)

(step :before_step
      [board scenario]
      board)

(defn the-datafied-table-should-be-foo-col-bar
      {:scene/step "the datafied table should be"}
      [board datafied-table]
      (assert (= (:table board)
                 (read-string datafied-table)))
      board)

File: jlc_ruby/example/belly.rb
# frozen_string_literal: true

require 'jukebox'

# Example tests using jukebox
module ExampleTests
  extend Jukebox

  step 'today is Sunday' do
    |board|
    puts '-> today is Sunday'
    board
  end

  step 'I ask whether it\'s Friday yet' do
    |board|
    puts '-> I ask whether it\'s Friday yet'
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step 'I should be told {string}', {resources: ["kafka/topic-h"]} do
    |board, _string|
    puts '-> I should be told {string}'
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step 'I have {int} cukes in my belly' do
    |board, cukes|
    p "-> I HAVE #{cukes} CUKES IN MY BELLY"
    board
  end

  step 'I wait {int} hour' do
    |board, _hours|
    puts '-> I wait {int} hour'
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step 'my belly should growl' do
    |board|
    puts '-> my belly should growl'
    # Write code here that turns the phrase above into concrete actions
    board
  end

  step :before, :after_step, tags: "@tag1 or @tag2 and @rb" do
    |board|
    board
  end

  step :before, :after do
    |board|
    board
  end

  step :before do
    |board|
    board
  end

  require 'jukebox/cukes'
  extend Jukebox::Cukes
  # Before do |board|
  #   board
  # end

  step 'a ruby step that fails' do |board|
    puts '=> a ruby step that fails'
    pending! # Write code here that turns the phrase above into concrete actions
    board # return the updated board
  end

  # TODO: Remove
  # step 'I have {float} cukes in my belly' do
  # |board|
  #   puts '=> I have {float} cukes in my belly'
  #   pending! # Write code here that turns the phrase above into concrete actions
  #   board # Return the updated board
  # end
  #
  # step 'I have this table' do
  #   |board|
  #   pending! # Write code here that turns the phrase above into concrete actions
  #   board # Return the updated board
  # end

  step 'the scenario should end with an error' do
    |board|
    # pending! # Write code here that turns the phrase above into concrete actions
    board # Return the updated board
  end

  # step 'a clojure step that fails' do
  #   |board|
  #   pending! # Write code here that turns the phrase above into concrete actions
  #   board # Return the updated board
  # end
  # step 'the datafied table should be' do
  # |board|
  #   pending! # Write code here that turns the phrase above into concrete actions
  #   board # Return the updated board
  # end
end
